src/monitoring_tests/combinatorial_auction_surplus_test.py

```diff
@@ -0,0 +1,266 @@
+        filtering. This takes care of the case of solvers providing a lot of surplus but at really
+        large costs.
+        """
+        trades_dict = self.get_uid_trades(solution)
+        surplus_dict: dict[tuple[str, str], Fraction] = {}
+        for uid in trades_dict:
+            trade = trades_dict[uid]
+            token_pair = (trade.data.sell_token.lower(), trade.data.buy_token.lower())
+
+            # compute the conversion rate of the surplus token to ETH
+            if trade.get_surplus_token() == trade.data.sell_token:
+                surplus_token_to_eth = Fraction(
+                    int(prices[token_pair[0]]),
+                    10**36,
+                )
+            elif trade.get_surplus_token() == trade.data.buy_token:
+                surplus_token_to_eth = Fraction(
+                    int(prices[token_pair[1]]),
+                    10**36,
+                )
+
+            surplus_dict[token_pair] = (
+                surplus_dict.get(token_pair, 0)
+                + surplus_token_to_eth * trade.get_surplus()
+            )
+
+        # use the minimum of surplus and objective in case there is only one token pair
+        if len(surplus_dict) == 1:
+            for token_pair in surplus_dict:
+                surplus_dict[token_pair] = min(
+                    surplus_dict[token_pair], solution["objective"]["total"]
+                )
+                # surplus_dict[token_pair] = solution["objective"]["total"]
+
+        return surplus_dict
+
+    def compute_baseline_surplus(
+        self, aggregate_solutions: list[dict[tuple[str, str], Fraction]]
+    ) -> dict[tuple[str, str], tuple[Fraction, int]]:
+        """Computes baseline surplus for all token pairs.
+        The baseline is computed from those solutions which only contain orders for a single token
+        pair.
+        The result is a dict containing directed token pairs and the aggregated surplus on them as
+        well as the index of the corresponding solution in the solutions list.
+        """
+        result: dict[tuple[str, str], tuple[Fraction, int]] = {}
+        for i, aggregate_solution in enumerate(aggregate_solutions):
+            if len(aggregate_solution) == 1:
+                token_pair, surplus = list(aggregate_solution.items())[0]
+                surplus_old = result.get(token_pair, (0, -1))[0]
+                if surplus > surplus_old:
+                    result[token_pair] = (surplus, i)
+        return result
+
+    def filter_solutions(
+        self,
+        aggregate_solutions: list[dict[tuple[str, str], Fraction]],
+        baseline_surplus: dict[tuple[str, str], tuple[Fraction, int]],
+    ) -> list[list[int]]:
+        """Filter out solutions which do not provide enough surplus.
+        Only solutions are considered for ranking that supply more surplus than any of the single
+        aggregate order solutions. The function returns a list of bools where False means the
```

Comment:
The comment about what the function returns is not accurate.

Also, maybe we could make the return of the function a bit easier to parse?

Reply:
I fixed the comment and added some rational for making it this complicated.
A simpler design would be nice though.